lib/promscrape/scrapework.go

```diff
@@ -422,7 +428,7 @@ func (sw *scrapeWork) scrapeInternal(scrapeTimestamp, realTimestamp int64) error
 	// is occupied during parsing of the read response body below.
 	// This also allows measuring the real scrape duration, which doesn't include
 	// the time needed for processing of the read response.
-	err := sw.ReadData(body)
+	err := sw.ReadData(body, &sw.contentType)
```

Comment:
Does stream parsing mode makes sense to the binary content WDYT? There is a check below for it
`if err == nil && sw.needStreamParseMode(len(body.B))`

Reply:
Also, 

 There is code below, that checks a number of added series. Looks like it only expects plain text format...
```go
seriesAdded := 0
	if !areIdenticalSeries {
		// The returned value for seriesAdded may be bigger than the real number of added series
		// if some series were removed during relabeling.
		// This is a trade-off between performance and accuracy.
		seriesAdded = sw.getSeriesAdded(lastScrape, bodyString)
	}
```

Reply:
> Does stream parsing mode makes sense to the binary content WDYT?
stream mode is used in oneshot processing as well in a part, which processes stale series

Reply:
updated code, which works with plaintext, it now partially parses protobuf as well